packages/salesforcedx-vscode-core/package.json

```diff
@@ -262,6 +258,22 @@
         {
           "command": "sfdx.debug.isv.bootstrap",
           "when": "config.salesforcedx-vscode-core.isv-debugger.enabled"
+        },
+        {
+          "command": "sfdx.force.apex.test.last.class.run",
+          "when": "sfdx:project_opened && sfdx:has_cached_test_class"
+        },
+        {
+          "command": "sfdx.force.apex.test.last.method.run",
+          "when": "sfdx:project_opened && sfdx:has_cached_test_method"
+        },
+        {
+          "command": "sfdx.force.apex.test.class.run",
```

Comment:
I want this to be usable as a "backing" function but I don't want this to show up in the command palette. So I've making sure to use `false`.

Reply:
We want these in the command palette, right?  When the context is set.  W/ just `false`, I'm not seeing that "SFDX: Re-Run Last..." appears in the "recently used" section of the command palette.  Also, w/ `false`, the following is no longer needed...

https://github.com/forcedotcom/salesforcedx-vscode/blob/develop/packages/salesforcedx-vscode-core/src/commands/forceApexTestRunCodeAction.ts#L100

The above applies to `sfdx.force.apex.test.method.run` to `false` too.

Reply:
I created new commands for the "last executed" ones. These ones shouldn't show up in the palette for now. Commands can exist (and be invokable) even if they don't show up in the palette. I've used that pattern before. Or, is there a reason why you want these to show up? 

Hmmm. I've tested and I do see the command show up. Here's the gif:

![test](https://user-images.githubusercontent.com/54414/36459022-947662ae-1666-11e8-84e5-5fd640d9f1ef.gif)

Reply:
The main reason I did this was because all test commands will delegate to the `sfdx.force.apex.test.class.run` method. 

For the codeAction, it delegates first to the `forceApexTestClassRunCodeActionDelegate` action which will set the context.

This, then enables the `sfdx.force.apex.test.last.class.run` command, which also delegates to the actual `sfdx.force.apex.test.class.run` command.

Yes, we now have more layers on indirection. The reason I did it this way was because we using the label "Re-Run Last Invoked Apex Test" on the `sfdx.force.apex.test.class.run` which is a more general command than just re-running the last test.

We might want to re-use the general purpose `sfdx.force.apex.test.class.run` in the future so I didn't want to just conflate it with running the last test.